### ISSUE TITLE
Elementary OS: menu items misplaced after refresh/change

### ISSUE THREAD

**Issue by megatux** [yktoo/indicator-sound-switcher#55]
![captura realizada el 2017-06-15 09 24 19](https://user-images.githubusercontent.com/850020/27181095-7024fe9e-51ac-11e7-9e47-552f55f00237.png)

**Comment by megatux** [comment]
Also, if I click "Refresh" all inputs & outputs move to the bottom, too.

**Comment by yktoo** [comment]
Was already reported as #44 but no further info provided.

It looks like GTK+ bug and I wasn't able to reproduce it. What's your OS version?

Can you insert
```python
print('Label: "%s"; index: "%d"' % (label, i))
```
before `self.menu.insert(new_item, i)` in `indicator-sound-switcher` and attach the output when it's run with the `-vv` command line switch, both before and after the refresh?

**Comment by megatux** [comment]
Elementary OS Loki with GTK+ 3.18.9.  I did a refresh first, then a plug/unplug of headset. Attached imgs & debug output:

![s1](https://user-images.githubusercontent.com/850020/29782547-67798fcc-8bf3-11e7-8a9e-b43836ee7ff8.png)

![s2](https://user-images.githubusercontent.com/850020/29782555-6a33ce1c-8bf3-11e7-88e0-f8aa8b8f2cac.png)
[debug.txt](https://github.com/yktoo/indicator-sound-switcher/files/1257470/debug.txt)

Let me know if you need something else. Thx!

**Comment by yktoo** [comment]
This is apparently Elementary specific as it's never been reported with Ubuntu. Will look at it as soon as I have time.

**Comment by davidmaxwaterman** [comment]
I think I see this on ubuntu 17:10 too:

![screenshot from 2018-02-23 17-28-21](https://user-images.githubusercontent.com/975698/36587152-377d8b08-18bf-11e8-929e-cecd0cf20ae5.png)

It looks fine when I start the app, but when I click refresh, all the outputs are moved under the inputs.

**Comment by davidmaxwaterman** [comment]
Any thoughts on this? It's still broken many months later, and I'm not sure if it is a proven problem or not.

**Comment by yktoo** [comment]
It must be a bug in Gtk. If I look at the item order in memory, it's perfectly correct:
```
######### menu_insert_ordered_item(Outputs, ,     High Definition Audio Controller ‣ HDMI / DisplayPort, False)
#########+++ idx_from=3, idx_to=3
---------------- Item [2]/'Outputs'
---------------- Item [3]/''
------------ Added item 3/'    High Definition Audio Controller ‣ HDMI / DisplayPort'
######### menu_insert_ordered_item(Outputs, ,     High Definition Audio Controller ‣ HDMI / DisplayPort 2, False)
#########+++ idx_from=3, idx_to=4
---------------- Item [2]/'Outputs'
---------------- Item [3]/'    High Definition Audio Controller ‣ HDMI / DisplayPort'
---------------- Item [4]/''
------------ Added item 4/'    High Definition Audio Controller ‣ HDMI / DisplayPort 2'
```

Yet, the *Outputs* item is rendered *below* the corresponding output item:
![screenshot from 2018-04-11 15-54-42](https://user-images.githubusercontent.com/1675911/38621210-aa6c28ee-3da0-11e8-817d-4630485d8bde.png)

**Comment by yktoo** [comment]
Resolved in 2.2.0.
